### PR TITLE
Update Developer Guide

### PR DESCRIPTION
**Pull request by hsiaotingluv**
updated
1. Logic
2. Remark implementation
3. Deletem implementation
4. Sort implementation

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -212,13 +196,28 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
 
 The features mentioned are:
-- [Adding contacts with optional arguments](#add-contacts-with-optional-arguments)
-- [Finding contacts by keywords](#find-contacts-by-keywords)
-- [Deleting multiple contacts by keywords](#delete-by-keywords)
-- [Importing a JSON file](#import-json-file)
-- [Undoing / redoing command](#proposed-undoredo-feature)
-- [Input Suggestion](#input-suggestion)
-- {and so on...}
+1. Viewing help
```
Comment (bklimey) [inline docs/DeveloperGuide.md:199]: > This section describes some noteworthy details on how certain features are implemented.

Perhaps we may not want to describe all the implementations?

**Review by bklimey — COMMENTED** [review]
Minor comment

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (hsiaotingluv) [reply docs/DeveloperGuide.md:199]: yes I am listing them all here first just in case any wants to add, but we may delete later

**Review by kevinmingtarja — APPROVED** [review]
LGTM!

**Pull request merged**